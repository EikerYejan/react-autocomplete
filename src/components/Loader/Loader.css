.loader {
  border: 3px solid var(--white); /* Light grey */
  border-top: 3px solid var(--blue); /* Blue */
  border-radius: 50%;
  width: 24px;
  height: 24px;
  animation: spin 0.8s linear infinite;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
