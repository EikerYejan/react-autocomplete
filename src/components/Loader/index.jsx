import React from 'react'
import './Loader.css'

class Loader extends React.Component {
  render() {
    return (
      <div {...this.props} className={`loader ${this.props?.className}`}>
        <div />
        <div />
      </div>
    )
  }
}

export default Loader
