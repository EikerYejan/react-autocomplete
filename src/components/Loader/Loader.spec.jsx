import React from 'react'
import { render } from '@testing-library/react'
import Loader from '.'

describe('<Loader />', () => {
  it('Should render', () => {
    const { baseElement } = render(<Loader />)

    expect(baseElement).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()
  })
})
