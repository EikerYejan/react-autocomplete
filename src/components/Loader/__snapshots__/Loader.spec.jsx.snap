// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Should render 1`] = `
<body>
  <div>
    <div
      class="loader undefined"
    >
      <div />
      <div />
    </div>
  </div>
</body>
`;
